libres/lib/analysis/update.cpp

```diff
@@ -0,0 +1,680 @@
+   this also reflects on the degree of change of the various members.
+
+     One ministep: This corresponds to one full execution of the function
+         enkf_main_analysis_update().
+
+     One dataset: Each ministep can consist of several local dataset.
+
+     One node: Each local dataset will consist of several nodes.
+
+   The members explicitly marked with a mutable: comment will vary in the
+   lifetime of the serialization_info, the other members will be constant. 
+*/
+
+typedef struct {
+    int row_offset;
+    const active_list_type *active_list;
+    const char *key;
+} serialize_node_info_type;
+
+typedef struct {
+    enkf_fs_type *src_fs;
+    enkf_fs_type *target_fs;
+    const ensemble_config_type *ensemble_config;
+    int iens1; /* Inclusive lower limit. */
+    int iens2; /* NOT inclusive upper limit. */
+    int report_step;
+    int target_step;
+    run_mode_type run_mode;
+    matrix_type *A;
+    const int_vector_type *iens_active_index;
+
+    std::vector<int>
+        active_size; /* mutable: For the serialization of one dataset - many nodes */
+    std::vector<int>
+        row_offset; /* mutable: For the serialization of one dataset - many nodes */
+    serialize_node_info_type
+        *node_info; /* mutable: For the serialization of one node */
+} serialize_info_type;
+
+static int __get_active_size(const ensemble_config_type *ensemble_config,
+                             enkf_fs_type *fs, const char *key, int report_step,
+                             const active_list_type *active_list) {
+    const enkf_config_node_type *config_node =
+        ensemble_config_get_node(ensemble_config, key);
+    /*
+     This is very awkward; the problem is that for the GEN_DATA
+     type the config object does not really own the size. Instead
+     the size is pushed (on load time) from gen_data instances to
+     the gen_data_config instance. Therefor we have to assert
+     that at least one gen_data instance has been loaded (and
+     consequently updated the gen_data_config instance) before we
+     query for the size.
+  */
+    {
+        if (enkf_config_node_get_impl_type(config_node) == GEN_DATA) {
+            enkf_node_type *node = enkf_node_alloc(config_node);
+            node_id_type node_id = {.report_step = report_step, .iens = 0};
+
+            enkf_node_load(node, fs, node_id);
+            enkf_node_free(node);
+        }
+    }
+
+    {
+        active_mode_type active_mode = active_list_get_mode(active_list);
+        int active_size;
+        if (active_mode == INACTIVE)
+            active_size = 0;
+        else if (active_mode == ALL_ACTIVE)
+            active_size =
+                enkf_config_node_get_data_size(config_node, report_step);
+        else if (active_mode == PARTLY_ACTIVE)
+            active_size = active_list_get_active_size(active_list, -1);
+        else {
+            util_abort("%s: internal error .. \n", __func__);
+            active_size = -1; /* Compiler shut up */
```

Comment:
I tried compiling without this line and the compiler (clang at least) does not seem to care.
Perhaps this is some legacy stuff?